### DISCUSSION TITLE
Help with getting a VirtFusion based VPS working with Talos (QEMU)

### DISCUSSION THREAD

**Discussion by benosman** [siderolabs/talos#8547] [Q&A]
Hi all,

I'd like to add a simple VPS to my newly created OCI Talos cluster, but I'm having some issues.

Background: OCI based cluster is 3 Arm based nodes in control plane, and 1 worker node. They all communicated over private subnet, kubespan is enabled, and there is a load balancer pointing at the talos and kubernetes APIs. That part is all good.

I want to add an external VPS node to the cluster, it is provided by an independent web host, and they use the [VirtFusion](https://virtfusion.com) QEMU Virtualization platform. It is pretty rudimentary, I can pick from a choice of linux os, boot from a preselected or custom ISO.

There is no DHCP server, there is however a RO virtual disk at /dev/sda that contains cloud init files - `meta-data`, `user-data` and `network-config`

### Approach 1. Install nocloud Talos variant 

Given that there is a disk with the label cidata, i figured I could use the nocloud variant and it could load the network config from that disk and it would work. So using the rescue image I flashed the primary disk with talos no cloud.

However, it doesn't seem to pick up the drive, and so when it boots it just fails with network unreachable.

![image](https://github.com/siderolabs/talos/assets/305317/d1c18d61-400c-467d-a74d-fd472cd71191)

![image](https://github.com/siderolabs/talos/assets/305317/71b12180-a244-4c90-afc7-a50ad099715f)

![image](https://github.com/siderolabs/talos/assets/305317/3ff90f7d-60a2-4fbb-b549-21bf4a41b40f)

So with this, I reached a dead end - is there a way to get Talos to load config from this drive?

### Approach 2. Install metal ISO Talos variant 

For this, I need to boot from the ISO and manually set the networking, then run machine config to install on to the drive.

I was having some issues where when it rebooted after configuration, or if it booted at any time without the ISO present it would fail to become ready. As I was going through it systematically to document it, it now seems to work, so for now I will go with this approach unless I start hitting that issue again.

I am curious though, why approach 1 doesn't work, and if it could be made to.

**Comment by smira** [comment]
If you have problem like that (network not configured), please submit full console logs. It is called 'serial console logs' usually in the UI.

**Reply by benosman** [reply]
Unfortunately, I can't see anything like that in my hosts UI. The view I have screenshotted is from the VNC plugin they offer, but that is it. 

Are these logs persistent between reboots?  I could try boot from the rescue image and retrieve them if so.

**Reply by smira** [reply]
No, but QEMU does support them for sure. If you have VNC, you can do Alt+F1, switch to the first console and scroll back the logs to see all of them on a bigger screen. But either way, if Talos fails to pick up nocloud, it would say so and why.
